power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/sparse_lu_solver.hpp

```diff
@@ -314,7 +315,7 @@ template <class Tensor, class RHSVector, class XVector> class SparseLUSolver {
                     // should always found
                     assert(found != col_indices.cbegin() + row_indptr[l_row + 1]);
                     assert(*found == u_col);
-                    a_idx = (Idx)std::distance(col_indices.cbegin(), found);
+                    a_idx = narrow_cast<Idx>(std::distance(col_indices.cbegin(), found));
```

Comment:
Why wont static cast be applicable here?

Reply:
because depending on the OS and architecture, it may be a useless cast and `clang` and `gcc` will complain about that. E.g.: `static_cast<int32_t>(int32_t{})`

In particular, MacOS had complaints about the result type of `std::distance`